lib/tar.ml

```diff
@@ -773,18 +819,31 @@ module HeaderWriter(Async: ASYNC)(Writer: WRITER with type 'a t = 'a Async.t) =
     Header.marshal ~level buffer header;
     really_write fd buffer
 
-  let write ?level header fd =
+  let write_extended ?level ~link_indicator hdr fd =
+    let link_indicator_name = match link_indicator with
+      | Header.Link.PerFileExtendedHeader -> "paxheader"
+      | Header.Link.GlobalExtendedHeader -> "pax_global_header"
+      | _ -> assert false
+    in
+    let pax_payload = Header.Extended.marshal hdr in
+    let pax = Header.make ~link_indicator link_indicator_name
+                (Int64.of_int @@ Cstruct.length pax_payload) in
+    write_unextended ?level pax fd
+    >>= fun () ->
+    really_write fd pax_payload
+    >>= fun () ->
+    really_write fd (Header.zero_padding pax)
+
+  let write ?level ?global header fd =
```

Comment:
After reading this again I'm not sure I like the `?global` argument. It's unclear if this is to keep track of the global extended header state (which it is not), and passing the same `global` value would lead to writing the same global extended header multiple times. This interface also doesn't allow writing an "empty" tar archive with just a comment, for example.

I think I would prefer another function `write_global` that takes a `Tar.Header.Extended.t` and does the required serializations and writes it to output.